geopandas/tools/tests/test_sjoin.py

```diff
@@ -200,6 +212,17 @@ def test_sjoin_duplicate_column_name(self):
         assert 'Shape_Area_left' in df.columns
         assert 'Shape_Area_right' in df.columns
 
+    @pytest.mark.parametrize('how', ['left','right','inner'])
+    def test_sjoin_named_index(self,how):
+        #original index names should be unchanged
+        pointdf2 = self.pointdf.copy()
+        pointdf2.index.name = 'pointid'
+        df = sjoin(pointdf2, self.polydf, how=how)
+        assert pointdf2.index.name == 'pointid'
+        assert self.polydf.index.name == None
+        #output index should be unamed since it is a new index?
+        #assert that extra index is called index_lhs or index_rhs?
```

Comment:
I'd say not necessary to test either of those things (and so would remove the comments).

Reply:
It might not be necessary for this pull request but I think that how sjoin does the naming of the index in the output should be explicitly specified and documented

Reply:
Well, I was thinking that those might be better addressed in #498, which might change the behaviors you mentioned. I don't see harm in adding those tests, even if the behavior is going to change, but I didn't see those tests as necessary to confirm that the changes you made addressed the problem it was aiming to solve.